src/mgr/ActivePyModules.cc

```diff
@@ -369,6 +369,34 @@ PyObject *ActivePyModules::get_python(const std::string &what)
         pg_map.dump_pool_stats_full(osd_map, nullptr, &f, true);
       });
     return f.get();
+  } else if (what == "pg_stats") {
+    cluster_state.with_pgmap(
+        [&f, &tstate](const PGMap &pg_map) {
+      PyEval_RestoreThread(tstate);
+      pg_map.dump_pg_stats(&f, false);
+    });
+    return f.get();
+  } else if (what == "pool_stats") {
+    cluster_state.with_pgmap(
+        [&f, &tstate](const PGMap &pg_map) {
+      PyEval_RestoreThread(tstate);
+      pg_map.dump_pool_stats(&f);
+    });
+    return f.get();
+  } else if (what == "pg_ready") {
+#if 0
+     // with_server doesn't exist and we don't need locking for an atomic
+    cluster_state.with_server(
+        [&f, &tstate](const DaemonServer &server) {
+      PyEval_RestoreThread(tstate);
+      server.dump_pg_ready(&f);
+
+    });
```

Comment:
@tchaikov @neha-ojha @jdurgin There is no with_server().  I added because pgmap_ready is in the DaemonServer class and it seemed to be how the other dump requests work.   The value pgmap_ready is an atomic so no locking is required.  I don't understand what the cluster_state.with_xxx() is really doing.

Reply:
i think the idea is to guard whatever code which tries to access the cluster state with the `ClusterState::lock`.

Reply:
agreed, there's no need for a similar function here since you're just accessing an atomic